[BUG] Static Code Analysis findings

I used the one line scan tool to create a list of findings for cppcheck and Infer. The two tools produce a bunch of issues.

**Describe the bug**
The below list of code analysis findings should be analyzed, and eventually addressed.

**To Reproduce**
Run the following (assuming infer and cppcheck are present)
```
https://github.com/awslabs/one-line-scan.git
./one-line-scan -o infer_files --trunc-existing --no-gotocc --infer -- make -B
./one-line-scan -o cppcheck_files --trunc-existing --no-gotocc --cppcheck -- make -B
```

**Expected behavior**
All enumerated bugs are false positives.

**Additional context**
I filtered cppchecks shift and overflow warnings.

## List of findings

```
arch/x86/boot/multiboot.c:102: (style:variableScope) The scope of the variable 'avail' can be reduced.
arch/x86/boot/multiboot.c:113: (portability:arithOperationsOnVoidPointer) 'r->start' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
arch/x86/boot/multiboot.c:126: (portability:arithOperationsOnVoidPointer) 'r->start' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
arch/x86/traps.c:110: [ERROR] NULL_DEREFERENCE pointer `percpu` last assigned on line 106 could be null and is dereferenced at line 110, column 5.
common/acpi.c:79: (portability:arithOperationsOnVoidPointer) 'addr' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
common/acpi.c:117: (portability:arithOperationsOnVoidPointer) 'kmap(mfn,0,1|2|32|64)' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
common/acpi.c:123: [ERROR] NULL_DEREFERENCE pointer `rsdt` last assigned on line 121 could be null and is dereferenced at line 123, column 27.
common/acpi.c:135: [ERROR] NULL_DEREFERENCE pointer `xsdt` last assigned on line 133 could be null and is dereferenced at line 135, column 27.
common/acpi.c:160: [ERROR] NULL_DEREFERENCE pointer `madt` last assigned on line 157 could be null and is dereferenced at line 160, column 58.
common/acpi.c:162: (portability:arithOperationsOnVoidPointer) '(void*)(unsigned long)(madt)' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
common/acpi.c:163: (portability:arithOperationsOnVoidPointer) 'addr' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
common/console.c:83: (style:funcArgNamesDifferent) Function 'register_console_callback' argument 1 names different: declaration 'func' definition 'cb'.
common/sched.c:163: [ERROR] INTEGER_OVERFLOW_L2 ([0, +oo] - 1):unsigned32.
include/arch/x86/pagetable.h:230: [ERROR] NULL_DEREFERENCE pointer `l2e` last assigned on line 228 could be null and is dereferenced at line 230, column 24.
lib/string.c:206: (style:variableScope) The scope of the variable 'c' can be reduced.
mm/pmm.c:90: (portability:arithOperationsOnVoidPointer) 'r->end' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
mm/pmm.c:90: (portability:arithOperationsOnVoidPointer) 'r->start' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
mm/slab.c:235: (style:redundantAssignment) Variable 'alloc_pages' is reassigned a value before the old one has been used.
smp/mptables.c:87: (portability:arithOperationsOnVoidPointer) 'sysm_addr' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
include/arch/x86/apic.h:83: (portability:arithOperationsOnVoidPointer) '(void*)(unsigned long)(4276092928U)' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
include/arch/x86/apic.h:87: (portability:arithOperationsOnVoidPointer) '(void*)(unsigned long)(4276092928U)' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
include/mm/vmm.h:48: (portability:arithOperationsOnVoidPointer) 'get_free_pages(order,flags)' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
include/mm/vmm.h:52: (portability:arithOperationsOnVoidPointer) 'get_free_page(flags)' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
```

I just re-run the analysis on the current HEAD commit: "0b8c2c758efadbd95db84eacec700bb69315ec0e (HEAD, origin/mainline, origin/HEAD) Add docker build test workflow"

```
arch/x86/boot/multiboot.c:102: (style:variableScope) The scope of the variable 'avail' can be reduced.
arch/x86/boot/multiboot.c:113: (portability:arithOperationsOnVoidPointer) 'r->start' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
arch/x86/boot/multiboot.c:126: (portability:arithOperationsOnVoidPointer) 'r->start' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
arch/x86/traps.c:111: [ERROR] NULL_DEREFERENCE pointer `percpu` last assigned on line 107 could be null and is dereferenced at line 111, column 5.
common/acpi.c:79: (portability:arithOperationsOnVoidPointer) 'addr' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
common/acpi.c:117: (portability:arithOperationsOnVoidPointer) 'kmap(mfn,0,1|2|32|64)' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
common/acpi.c:173: (portability:arithOperationsOnVoidPointer) '(void*)(unsigned long)(madt)' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
common/acpi.c:174: (portability:arithOperationsOnVoidPointer) 'addr' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
common/console.c:83: (style:funcArgNamesDifferent) Function 'register_console_callback' argument 1 names different: declaration 'func' definition 'cb'.
common/sched.c:186: [ERROR] INTEGER_OVERFLOW_L2 ([0, +oo] - 1):unsigned32.
common/setup.c:70: (style:variableScope) The scope of the variable 'optval' can be reduced.
common/setup.c:70: (style:variableScope) The scope of the variable 'q' can be reduced.
common/symbols.c:33: (portability:arithOperationsOnVoidPointer) 'symbol_addresses[index]' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
common/symbols.c:74: [ERROR] INTEGER_OVERFLOW_L2 ([0, +oo] - 1):unsigned32 by call to `symbol_index_by_address`.
common/symbols.c:86: [ERROR] INTEGER_OVERFLOW_L2 ([0, +oo] - 1):unsigned32 by call to `symbol_index_by_address`.
include/arch/x86/apic.h:83: (portability:arithOperationsOnVoidPointer) '(void*)(unsigned long)(4276092928U)' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
include/arch/x86/apic.h:87: (portability:arithOperationsOnVoidPointer) '(void*)(unsigned long)(4276092928U)' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
include/arch/x86/pagetable.h:232: [ERROR] NULL_DEREFERENCE pointer `l2e` last assigned on line 230 could be null and is dereferenced at line 232, column 12.
include/mm/vmm.h:48: (portability:arithOperationsOnVoidPointer) 'get_free_pages(order,flags)' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
include/mm/vmm.h:52: (portability:arithOperationsOnVoidPointer) 'get_free_page(flags)' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
include/string.h:118: (portability:arithOperationsOnVoidPointer) 'd' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
include/string.h:118: (portability:arithOperationsOnVoidPointer) 's' is of type 'const void *'. When using void pointers in calculations, the behaviour is undefined.
include/string.h:183: [ERROR] UNINITIALIZED_VALUE The value read from res was never initialized.
include/string.h:320: (style:redundantAssignment) Variable 'b' is reassigned a value before the old one has been used.
include/string.h:353: (style:variableScope) The scope of the variable 'c' can be reduced.
include/string.h:356: (style:redundantAssignment) Variable 's' is reassigned a value before the old one has been used.
include/string.h:378: (style:variableScope) The scope of the variable 'c1' can be reduced.
include/string.h:378: (style:variableScope) The scope of the variable 'c2' can be reduced.
include/string.h:397: (style:variableScope) The scope of the variable 'c1' can be reduced.
include/string.h:397: (style:variableScope) The scope of the variable 'c2' can be reduced.
lib/string.c:206: (style:variableScope) The scope of the variable 'c' can be reduced.
mm/pmm.c:93: (portability:arithOperationsOnVoidPointer) 'r->start' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
mm/pmm.c:94: (portability:arithOperationsOnVoidPointer) 'r->end' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
mm/slab.c:235: (style:redundantAssignment) Variable 'alloc_pages' is reassigned a value before the old one has been used.
smp/mptables.c:87: (portability:arithOperationsOnVoidPointer) 'sysm_addr' is of type 'void *'. When using void pointers in calculations, the behaviour is undefined.
toolkit/cache/lib.c:46: [ERROR] UNINITIALIZED_VALUE The value read from t[_] was never initialized.
toolkit/cache/lib.c:55: [ERROR] UNINITIALIZED_VALUE The value read from t[_] was never initialized.
```

--- a/.github/workflows/one-line-cr-bot.yml
+++ b/.github/workflows/one-line-cr-bot.yml
@@ -1,0 +1,80 @@
+# Author: Norbert Manthey <nmanthey@amazon.de>
+#
+# This workflow will present introduced defects of a pull request to a given
+# branch of a package.
+#
+# The workflow has locations labeled '[ACTION REQUIRED]' where adaptation for
+# your build might be required, as well as where to compare the findings to.
+#
+# To learn more about the available options, check the CLI parameters of the
+# script 'one-line-cr-bot.sh' in https://github.com/awslabs/one-line-scan.git
+name: One Line CR Bot
+
+on:
+  pull_request:
+    # [ACTION REQUIRED] Set the branch you want to analyze PRs for
+    branches: [ mainline ]
+
+  # [ACTION REQUIRED] Use this, if you want analysis for push to repository as well
+  push:
+    branches: [ mainline ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    # Get the code, fetch the full history to make sure we have the compare commit as well
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    # one-line-cr-bot.sh will get infer and cppcheck, if not available
+    - name: Install CppCheck Package
+      env:
+        # This is needed in addition to -yq to prevent apt-get from asking for user input
+        DEBIAN_FRONTEND: noninteractive
+      # [ACTION REQUIRED] Add your build dependencies here, drop cppcheck to get latest cppcheck
+      run: |
+        sudo apt-get install -y cppcheck
+
+    # Get the compare remote
+    - name: Setup Compare Remote
+      # [ACTION REQUIRED] Add the https URL of your repository
+      run: git remote add compare https://github.com/awslabs/ktf.git
+    - name: Fetch Compare Remote
+      run: git fetch compare
+
+    # Get one-line-scan, the tool we will use for analysis
+    - name: Get OneLineScan
+      run:  git clone -b one-line-cr-bot https://github.com/nmanthey/one-line-scan.git ../one-line-scan
+
+    # Check how repository is setup
+    - name: Be Verbose about Git Setup
+      run: |
+        git remote -v
+        git branch -a
+        git log --pretty=oneline --decorate --graph | head -n 10
+
+    # Run the analysis, parameterized for this package
+    - name: one-line-cr-analysis
+      env:
+        # [ACTION REQUIRED] Adapt the values below accordingly
+        # 'compare' is the name of the remote to use
+        BASE_COMMIT: "compare/mainline"
+        BUILD_COMMAND: "make -B all"
+        CLEAN_COMMAND: "make clean"
+        # Parameters to be forwarded to used tools in one-line-scan for customization
+        # Additional CppCheck parameters, do not use e.g. --inconclusive
+        CPPCHECK_EXTRA_ARG: "--enable=style --enable=performance --enable=information --enable=portability"
+        # Additional Infer parameters, do not use e.g. --pulse
+        INFER_ANALYSIS_EXTRA_ARGS: "--bufferoverrun"
+        # These settings are more preferences, and not directly related to your project
+        # Set INSTALL_MISSING to false, if ALL targetted tools are already present
+        INSTALL_MISSING: true
+        OVERRIDE_ANALYSIS_ERROR: true
+        REPORT_NEW_ONLY: true
+        VERBOSE: 0 # >0 shows all currently present defects as well
+      # Be explicit about the tools to be used
+      run: ../one-line-scan/one-line-cr-bot.sh -E infer -E cppcheck

